apps/web/pages/api/auth/[...nextauth].tsx

```diff
@@ -82,17 +95,19 @@ const providers: Provider[] = [
         throw new Error(ErrorCode.IncorrectUsernamePassword);
       }
 
-      if (user.identityProvider !== IdentityProvider.CAL) {
+      if (user.identityProvider !== IdentityProvider.CAL && !credentials.totpCode) {
         throw new Error(ErrorCode.ThirdPartyIdentityProviderEnabled);
       }
 
-      if (!user.password) {
+      if (!user.password && user.identityProvider !== IdentityProvider.CAL && !credentials.totpCode) {
```

Comment:
Same thing here, when a TOTP code is used in the login form, a password is not entered as there is no password because it's an external provider, so unless there is a totp code sent and we are in the presence of an external provider, we let the flow continue not throwing an error to be able to check the totp code